public/app/plugins/datasource/grafana-azure-monitor-datasource/azure_log_analytics/azure_log_analytics_datasource.test.ts

```diff
@@ -125,22 +125,38 @@ describe('AzureLogAnalyticsDatasource', () => {
     beforeEach(() => {
       datasourceRequestMock.mockImplementation((options: { url: string }) => {
         expect(options.url).toContain('metadata');
-        return Promise.resolve({ data: FakeSchemaData.getlogAnalyticsFakeMetadata(), status: 200 });
+        return Promise.resolve({ data: FakeSchemaData.getlogAnalyticsFakeMetadata(), status: 200, ok: true });
       });
     });
 
-    it('should return a schema with a table and rows', () => {
-      return ctx.ds.azureLogAnalyticsDatasource.getSchema('myWorkspace').then((result: KustoSchema) => {
-        expect(Object.keys(result.Databases.Default.Tables).length).toBe(2);
-        expect(result.Databases.Default.Tables.Alert.Name).toBe('Alert');
-        expect(result.Databases.Default.Tables.AzureActivity.Name).toBe('AzureActivity');
-        expect(result.Databases.Default.Tables.Alert.OrderedColumns.length).toBe(69);
-        expect(result.Databases.Default.Tables.AzureActivity.OrderedColumns.length).toBe(21);
-        expect(result.Databases.Default.Tables.Alert.OrderedColumns[0].Name).toBe('TimeGenerated');
-        expect(result.Databases.Default.Tables.Alert.OrderedColumns[0].Type).toBe('datetime');
-
-        expect(Object.keys(result.Databases.Default.Functions).length).toBe(1);
-        expect(result.Databases.Default.Functions.Func1.Name).toBe('Func1');
+    it('should return a schema to use with monaco-kusto', () => {
+      return ctx.ds.azureLogAnalyticsDatasource.getKustoSchema('myWorkspace').then((result: any) => {
```

Comment:
is there a reason why we don't use async/await here instead of returning a promise?

Reply:
Only just because "that's how it was before".

You're right though, it's better with a sink